src/lsp4clj/server.clj

```diff
@@ -245,3 +253,29 @@
   (chan-server (assoc opts
                       :input-ch (json-rpc/input-stream->input-chan in)
                       :output-ch (json-rpc/output-stream->output-chan out))))
+
+(defn init-socket-server [{:keys [address port]}]
+  (json-rpc/start-socket-server {:address address
+                                 :port (if (string? port) (parse-long port) port)}))
+
+#_{:clj-kondo/ignore [:clojure-lsp/unused-public-var]}
+(defn socket-server
+  "Starts listening on the address (loopback by default) and port. Blocks until
+  a client establishes a connection, then returns a chan-server which
+  communicates over the socket."
+  [{:keys [_address _port] :as opts}]
```

Comment:
Any reason for this destructuring other than making explicit what opts receives?

Reply:
No reason other than to be clear about the opts. The docstring mentions address and port, so I wanted to make the arg list match. I think the "correct" way to do this is to attach `:arglists` metadata, but I never do that because of a combination of laziness and fear that it'll get out of date. 😅

Reply:
The latest refactoring uses `:address` and `:port`, so the destructuring actually has a point now. :)